spec-main/api-session-spec.js

```diff
@@ -87,15 +87,26 @@ describe('session module', () => {
       expect(cs.some(c => c.name === name && c.value === value)).to.equal(true)
     })
 
-    it('yields an error when setting a cookie with missing required fields', async () => {
-      await expect((async () => {
-        const { cookies } = session.defaultSession
-        const name = '1'
-        const value = '1'
+    it('yields an error when setting a cookie with missing required fields', () => {
+      const { cookies } = session.defaultSession
+      const name = '1'
+      const value = '1'
+
+      expect((async () => {
```

Comment:
you need to `await` the result of this `expect`, otherwise the test will complete without waiting for the result of the assert. (which means the test function will need to become `async` again, too)
```suggestion
      await expect((async () => {
```

Reply:
@nornagon i'm pretty sure we can still remove an excess async by moving the primary one to the top-level like so?
```
it('yields an error when setting a cookie with missing required fields', async () => {
   const { cookies } = session.defaultSession
   const name = '1'
   const value = '1'
   expect(() => {
        await cookies.set({ url: '', name, value })
    }).to.eventually.be.rejectedWith('Failed to get cookie domain')
})
```

Reply:
Actually you could condense this down a little, because in general, 
```js
(async () => await foo(...))
// is equivalent to
(foo(...))
```
since `foo()` just returns a promise.

so in this case, we can write:

```js
await expect(cookies.set({...})).to.eventually.be.rejectedWith(...)
```